src/server/frontend_wayland/desktop_file_manager.cpp

```diff
@@ -27,6 +27,10 @@ namespace mf = mir::frontend;
 namespace
 {
 const char* DESKTOP_FILE_POSTFIX = ".desktop";
+
+const std::map<std::string, std::string> app_id_to_desktop_file_quirks = {
```

Comment:
We use east-const around here. (I note the line above doesn't, nor does it follow our naming style - but that's pre-existing.)

Also, I doubt that we will ever have enough quirks to justify a map. (Also a shame it can't be `constexpr`!)

I think
```suggestion
std::pair<std::string, std::string> const app_id_to_desktop_file_quirks[] = {
```
would be cleaner and more efficient.